### PR TITLE
KOGITO-5157: Integrate import-java-classes component into DMN Editor

### PR DESCRIPTION
**Pull request by yesamer**
https://issues.redhat.com/browse/KOGITO-5157

Showcase: https://yesamer.github.io/kogito-editors-java/

Screencast from 12-15-2021 08_21_34 PM

This PR integrates the new Import Java Classes Component inside the DMN Editor
To test it, please use the automatically generated `vsix` file by the `CI :: Full Downstream Build / build ` - to find it, go in Checks tab and see in the right part of the screen. You should find an artifact with the zip file containing the `vsix`

Points to consider:
- The LSP extension is currently MOCKED. This is a temporary solution until I integrate the component with the real service available here: https://github.com/kiegroup/kogito-tooling/pull/647.
- The mock works only with the following Keys (`Boo`, `Boom` `Book` and `Author`)
- The button is enabled for testing purposes. At the moment we need to release a new version of the editor, I will provide a PR to disable it (There's no reason to release a public version with a mock)
- If you try to reimport the same java class multiple times, the new imported class will replace the old one. I'm aware this is not a proper solution, so I'll improve this part in a future PR

### CODE REVIEW

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/types/jsinterop/JavaField.java
Comment (vpellegrino) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/types/jsinterop/JavaField.java:44]: Missing newline

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/types/listview/DataTypeList.java
Comment (vpellegrino) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/types/listview/DataTypeList.java:545]: this can be `final`

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/types/listview/DataTypeListView.java
Comment (vpellegrino) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/types/listview/DataTypeListView.java:429]: Please, help me to understand better why you need these two methods.

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/editors/types/listview/DataTypeListTest.java
Comment (vpellegrino) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/editors/types/listview/DataTypeListTest.java:881]: This test method is too long in my opinion. Maybe you could refactor it a bit, by separating in different methods the initialization, the action, and the verification.
WDYT?

**Review by vpellegrino — COMMENTED** [review]
I just performed a review of the code. 
I shared a couple of suggestions.
A big step forward for import-java-classes!
Thanks @yesamer

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/editors/types/listview/DataTypeListTest.java
Comment (yesamer) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/editors/types/listview/DataTypeListTest.java:881]: @vpellegrino Sure thing.

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/types/listview/DataTypeListView.java
Comment (yesamer) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/types/listview/DataTypeListView.java:429]: @vpellegrino That is a common solution already used in this repo to verify a static method is called. In this way, you can verify a static method is called without mocking it. Of course, It is possible to mock the static method using PowerMockito or the latest Mockito API, but that requires correctly setting up all required preconditions (eg. Annotations) but I preferred to use this indirection solution to simplify the testing process.  Aware is not ideal to modify the regular code to handle tests - so I can try to reconsider using the api. I'll let u know.

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/types/listview/DataTypeList.java
Comment (yesamer) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/types/listview/DataTypeList.java:545]: @vpellegrino and `final` will be

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/types/listview/DataTypeListView.java
Comment (vpellegrino) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/types/listview/DataTypeListView.java:429]: @yesamer fine for me to keep things as are.
No need to invest more time on this from my point of view.
Thanks for the clarification

**Review by domhanak — COMMENTED** [review]
I have finalized the review on related kogito-tooing PR, so I would wait with review here until it is merged . Is that ok with you @yesamer ?

**Comment by yesamer** [comment]
@domhanak I think you can consider them as unrelated because here I'm using a mocked of the services provided by the other pr. I'm currently working to integrate this component with the real service. 
Anyway, at this moment I'm would prefer to have a code review by engineers first, so I'll ping you when you can actually start the testing phase.

**Review by danielzhe — APPROVED** [review]

**Comment by yesamer** [comment]
Jenkins retest this

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/types/jsinterop/JavaField.java
Comment (yesamer) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/types/jsinterop/JavaField.java:45]: Done

**Comment by yesamer** [comment]
@tiagobento I applied all the suggestions you gave me about this component. Thank you for your availability and for the time spent on that. Can you please finalize the review? Thx :)

**Comment by yesamer** [comment]
jenkins retest this please

**Review by tiagobento — APPROVED** [review]
Thanks for applying the changes, Yeser! Code looks fine, we can make more improvements on the next PR if necessary!

**Comment by yesamer** [comment]
Hi @domhanak, this is finally ready to be reviewed. Let me know if you need more info. Thx u!

**Comment by vpellegrino** [comment]
@yesamer I will review this PR as well, probably tomorrow morning.

File: kogito-editors-js/packages/import-java-classes-component/src/components/ImportJavaClasses/Model/DMNSimpleType.ts
Comment (vpellegrino) [inline kogito-editors-js/packages/import-java-classes-component/src/components/ImportJavaClasses/Model/DMNSimpleType.ts:31]: For the future, I'd suggest that this kind of association + enum of DMN Simple types could come in input to the component from the GWT editor.
Because ideally, the list of types could change (unlikely but still possible), and generally talking, it's better to have the definitions only in one place (GWT editor) instead of duplicating them (component + GWT editor).
Indeed, for example, with KOGITO-4588 even if I start with an enum like you, I will follow the approach that I described above.

**Review by vpellegrino — APPROVED** [review]
Approving since code LGTM.
Just a minor observation about the approach.

File: kogito-editors-js/packages/import-java-classes-component/src/components/ImportJavaClasses/Model/DMNSimpleType.ts
Comment (yesamer) [reply kogito-editors-js/packages/import-java-classes-component/src/components/ImportJavaClasses/Model/DMNSimpleType.ts:31]: @vpellegrino Indeed, totally agree with you. Another reason for that change is: this component could be reused in BPMN editor too, so we need to parameterize that translation map in order to apply one for DMN and another one for BPMN.  I'll open a ticket to track it

**Comment by yesamer** [comment]
Hi @domhanak, kindly reminder. Can you please confirm you saw my previous message? Thank you.

**Comment by domhanak** [comment]
@yesamer I was on PTO, you can check the our calendar for example.

Downloaded the vsix, installed it, I can use the mock as described, please confirm the description is still valid before merging. I do not want to miss something due to my absence :) Thanks

**Review by domhanak — APPROVED** [review]

**Comment by yesamer** [comment]
Thank you @domhanak !

**Pull request merged**
